### PR TITLE
Fix Preset Tunnels

### PR DESCRIPTION
**Pull request by blueDev2**
## About the PR

I have no idea how to map, didn't know this would cause issues to mappers, sorry for that.

Now, to place a preset tunnel, place a RMCSpawnerXenoTunnel entity over the location where the preset tunnel should be. Set the tunnelName field to whatever the preset name should be. It must be unique.
## Requirements

- [X] I have read and am following the [Pull Request and Changelog Guidelines](https://docs.spacestation14.com/en/general-development/codebase-info/pull-request-guidelines.html).
- [X] I have added media to this PR or it does not require an ingame showcase.

## Breaking changes

**Changelog**
:cl:
- fix: Fixed preset tunnels needing renaming to properly work

### CODE REVIEW

File: Content.Shared/_RMC14/Xenonids/Construction/Tunnel/XenoTunnelSpawnerComponent.cs [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+/// For mapping purposes, the gamerule should use the information in this component to spawn tunnels with the proper name and hive
+/// </summary>
+[RegisterComponent]
+public sealed partial class XenoTunnelSpawnerComponent : Component
+{
+    [DataField]
+    public string? TunnelName;
+
+    [DataField]
+    public string? HiveName;
```
Comment (DrSmugleaf) [inline Content.Shared/_RMC14/Xenonids/Construction/Tunnel/XenoTunnelSpawnerComponent.cs:20]: This doesn't look used from the code, distress signal is the system creating the hive round-start anyways
```suggestion
```

File: Content.Server/_RMC14/Rules/CMDistressSignalRuleSystem.cs [hunk 2]
```diff
@@ -1332,10 +1335,20 @@ public void SetFriendlyHives(EntityUid hive)
...
+    public void SpawnTunnelsFromMarkers(EntityUid hive)
+    {
+        var tunnelSpawnerQuery = EntityQueryEnumerator<XenoTunnelSpawnerComponent>();
+        while (tunnelSpawnerQuery.MoveNext(out var uid, out var tunnelSpawnerComp))
         {
-            _hive.SetHive(uid, hive);
+            if (tunnelSpawnerComp.TunnelName is null)
+            {
+                continue;
+            }
```
Comment (DrSmugleaf) [inline Content.Server/_RMC14/Rules/CMDistressSignalRuleSystem.cs:1350]: This should ideally just generate the name on its own like 13 does, which uses the area name and coordinates:
`Southwest Abandoned Mining Caves (9, 16) Alpha`

**Review by DrSmugleaf — CHANGES_REQUESTED** [review]
Thanks for the PR!

**Review by DrSmugleaf — CHANGES_REQUESTED** [review]
Actually for this you should be able to just change the name of any tunnels found in distress signal rule system, you don't need a separate spawner

File: Content.Shared/_RMC14/Xenonids/Construction/Tunnel/XenoTunnelSpawnerComponent.cs (hunk 1 above)
Comment (blueDev2) [reply Content.Shared/_RMC14/Xenonids/Construction/Tunnel/XenoTunnelSpawnerComponent.cs:20]: It's supposed to be like, future proofed if there's multiple hives.
Though it's not very useful if they have randomly generated hive names.

**Comment by blueDev2** [comment]
> Actually for this you should be able to just change the name of any tunnels found in distress signal rule system, you don't need a separate spawner

How would I know what name each tunnel should be?
Edit: Nevermind, I misremembered parity, it's just default name.

**Comment by blueDev2** [comment]
Gonna make a new PR, It'll check all the currently placed tunnels during the game rule spawning, add them to the hive, and rename them to the default name.

**Pull request closed without merging**